internal/provider/vaultsecrets/dynamic_secret_aws.go

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
```

Comment:
The interfaces and abstractions may seem convoluted at first, but the benefit is that new secret types only need to add their new fields to the schema and implement the `dynamicSecret` or future `rotatingSecret` interface.